desktop/desktop
# Surface git config lock file errors to users and let them decide to delete

Closes #8675 

## Description

Recently we suspect some other semi-common tool has introduced a behavior which leaves Git configuration lock files lying around preventing us from writing to them.

Prior to this PR such errors would be silently suppressed and the user would have no idea why they couldn't advance past the configure git screen in the welcome flow or why they were unable to save their name and or email in preferences.

Now, when we detect this particular error we surface it to the user and let them decide how to proceed (with the option of deleting the lock file from within the app itself).

### Screenshots



<img width="1072" alt="image" src="https://user-images.githubusercontent.com/634063/72912653-3c773980-3d3c-11ea-80b0-b787a0354691.png">

<img width="1072" alt="image" src="https://user-images.githubusercontent.com/634063/72912684-4a2cbf00-3d3c-11ea-9f34-1098fb13afc7.png">

In the preferences view the user will be returned to the Git tab regardless of what tab they're on when they try to save.

## Testing

1. Create a lock file for the global configuration file (`touch ~/gitconfig.lock` on macOS/Linux)
2. Try to change your user name or email in preferences or in the welcome flow

Expected result: The new error message shows up and allows you to delete the file and try again.

## Release notes



Notes: [Improved] Show an error when writing configuration file fails due to existing lock file
